tgui/packages/tgui/interfaces/pda/pda_security.js

```diff
@@ -0,0 +1,148 @@
+import { useBackend, useLocalState } from "../../backend";
+import { createSearch } from 'common/string';
+import { flow } from 'common/fp';
+import { filter, sortBy } from 'common/collections';
+import { Box, Input, Button, Section, LabeledList } from "../../components";
+
+export const pda_security = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    records,
+  } = data;
+
+  return (
+    <Box>
+      {!records ? (
+        <SelectionView />
+      ) : (
+        <RecordView />
+      )}
+    </Box>
+  );
+};
+
+const SelectionView = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    recordsList,
+  } = data;
+
+  const [
+    searchText,
+    setSearchText,
+  ] = useLocalState(context, 'searchText', '');
+
+  // Search for peeps
+  const SelectMembers = (people, searchText = '') => {
+    const MemberSearch = createSearch(searchText, member => member.Name);
+    return flow([
+      // Null camera filter
+      filter(member => member?.Name),
+      // Optional search term
+      searchText && filter(MemberSearch),
+      // Slightly expensive, but way better than sorting in BYOND
+      sortBy(member => member.Name),
+    ])(recordsList);
+  };
+
+  const formattedRecords = SelectMembers(recordsList, searchText);
+
+  return (
+    <Box>
+      <Input
+        fluid
+        mb={1}
+        placeholder="Search records..."
+        onInput={(e, value) => setSearchText(value)} />
+      {formattedRecords.map(r => (
+        <Box key={r}>
+          <Button
+            content={r.Name}
+            icon="user"
+            onClick={() => act('Records', { target: r.ref })}
```

Comment:
[Use `[A.UID()]` over `\ref[A]`](https://github.com/ParadiseSS13/Paradise/blob/master/.github/CONTRIBUTING.md#use-auid-over-refa)

Reply:
Considering all the issues we have had recently with UID exhaustion, I am unsure of whether replacing all of these is a good idea. I am deferring to the other maints.